hibernate-core/src/main/java/org/hibernate/dialect/function/DateTruncEmulation.java

```diff
@@ -0,0 +1,160 @@
+				StandardFunctionReturnTypeResolvers.useArgType( 1 ),
+				StandardFunctionArgumentTypeResolvers.invariant( typeConfiguration, TEMPORAL, TEMPORAL_UNIT )
+		);
+		this.toDateFunction = toDateFunction;
+	}
+
+	@Override
+	public void render(
+			SqlAppender sqlAppender,
+			List<? extends SqlAstNode> sqlAstArguments,
+			SqlAstTranslator<?> walker) {
+		sqlAppender.appendSql( toDateFunction );
+		sqlAppender.append( '(' );
+		sqlAstArguments.get( 1 ).accept( walker );
+		sqlAppender.append( ',' );
+		sqlAstArguments.get( 2 ).accept( walker );
+		sqlAppender.append( ')' );
+	}
+
+	@Override
+	protected <T> SelfRenderingSqmFunction<T> generateSqmFunctionExpression(
+			List<? extends SqmTypedNode<?>> arguments,
+			ReturnableType<T> impliedResultType,
+			QueryEngine queryEngine,
+			TypeConfiguration typeConfiguration) {
+		final NodeBuilder nodeBuilder = queryEngine.getCriteriaBuilder();
+		final TemporalUnit temporalUnit = ( (SqmExtractUnit<?>) arguments.get( 1 ) ).getUnit();
+		final String pattern;
+		final String literal;
+		switch ( temporalUnit ) {
+			case YEAR:
+				pattern = "yyyy";
+				literal = "-01-01 00:00:00";
+				break;
+			case MONTH:
+				pattern = "yyyy-MM";
+				literal = "-01 00:00:00";
+				break;
+			case DAY:
+				pattern = "yyyy-MM-dd";
+				literal = " 00:00:00";
+				break;
+			case HOUR:
+				pattern = "yyyy-MM-dd HH";
+				literal = ":00:00";
+				break;
+			case MINUTE:
+				pattern = "yyyy-MM-dd HH:mm";
+				literal = ":00";
+				break;
+			case SECOND:
+				pattern = "yyyy-MM-dd HH:mm:ss";
+				literal = null;
+				break;
+			default:
+				throw new UnsupportedOperationException( "Temporal unit not supported [" + temporalUnit + "]" );
+		}
+		final SqmTypedNode<?> datetime = arguments.get( 0 );
+		final SqmExpression<?> formatExpression = queryEngine.getSqmFunctionRegistry()
+				.findFunctionDescriptor( "format" )
+				.generateSqmExpression(
+						asList(
+								datetime,
+								new SqmFormat(
+										pattern,
+										typeConfiguration.getBasicTypeForJavaType( String.class ),
+										nodeBuilder
+								)
+						),
+						null,
+						queryEngine,
+						typeConfiguration
+				);
+		final SqmExpression<?> formattedDatetime;
+		if ( literal != null ) {
+			formattedDatetime = queryEngine.getSqmFunctionRegistry()
+					.findFunctionDescriptor( "concat" )
+					.generateSqmExpression(
+							asList(
+									formatExpression,
+									new SqmLiteral<>(
+											literal,
+											typeConfiguration.getBasicTypeForJavaType( String.class ),
+											nodeBuilder
+									)
+							),
+							null,
+							queryEngine,
+							typeConfiguration
+					);
+		}
+		else {
+			formattedDatetime = formatExpression;
+		}
+		final SqmFormat sqmFormat = new SqmFormat(
+				"yyyy-MM-dd HH:mm:ss",
+				typeConfiguration.getBasicTypeForJavaType( String.class ),
+				nodeBuilder
+		);
+
+		return new SelfRenderingSqmFunction<>(
+				this,
+				this,
+				asList( datetime, formattedDatetime, sqmFormat ),
```

Comment:
Seems the first argument is only used in `SybaseDateTruncEmulation#render` method. Can you please add a comment here about this?